packages/0xcert-ethereum-xcert-contracts/src/contracts/xcert.sol

```diff
@@ -245,6 +210,57 @@ contract XcertToken is
     delete idToImprint[_tokenId];
   }
 
+  /**
+   * @dev Destroys a specified Xcert. Reverts if not called from Xcert owner or operator.
+   * @param _destroyer Address that is destroying the token.
+   * @param _tokenId Id of the Xcert we want to destroy.
+   * @param _feeToken The token then will be tranfered to the executor of this method.
+   * @param _feeValue The amount of token then will be tranfered to the executor of this method.
+   * @param _seed Arbitrary number to facilitate uniqueness of the order's hash. Usually timestamp.
+   * @param _expiration Timestamp of when the claim expires.
+   * @param _signature Data from the signature.
+   */
+  function destroyWithSignature(
+    address _destroyer,
+    uint256 _tokenId,
+    address _feeToken,
```

Comment:
Why does a fee need to be included here? The alternative is to handle fee payment with atomic swap.

Also this fee is a race condition. Miners will block all `destroyWithSignature` transactions and rewrite them from its own beneficiary address.

Reply:
Hum ... we didn't have plans to support `burn` in the ActionsOrder. Hum ... let me rethink this ...

Reply:
We can add feeRecipient which would prevent this. Because I dont believe actions like this fit into the ActionsOrder.